app/graphql/mutations/users/login_user.rb

```diff
@@ -0,0 +1,25 @@
+module Mutations
+  module Users
+    class LoginUser < ::Mutations::BaseMutation
+      null true 
+
+      argument :credentials, Types::AuthProviderCredentialsInput, required: false
+
+      field :token, String, null: true
+      field :user, Types::UserType, null: true
+
+      def resolve(credentials: nil)
+        #basic validation
+        return unless credentials
+        user = User.find_by username: credentials[:username]
+
+        return unless user
+        return unless user.authenticate(credentials[:password])
+        crypt = ActiveSupport::MessageEncryptor.new(Rails.application.credentials.secret_key_base.byteslice(0..31))
+        token = crypt.encrypt_and_sign("user-id:#{ user.id }")
```

Comment:
Should this be handled by `bcrypt`?

Reply:
I don't believe so. As I understand it `bcrypt` won't issue any tokens for login and `#authenticate` is what checks if the hashed password matches the `password_digest`.